maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -242,11 +242,12 @@
     private String[] additionalClasspathElements;
 
     /**
-     * The test source directory containing test class sources.
+     * The test source directory containing test class sources. Only used by the TestNG provider to process JavaDoc
+     * tags.
      *
      * @since 2.2
      */
-    @Parameter( defaultValue = "${project.build.testSourceDirectory}", required = true )
+    @Parameter( defaultValue = "${project.build.testSourceDirectory}", readonly = true )
     private File testSourceDirectory;
```

Comment:
Not sure about this

Reply:
@eolivelli 
Feel free to ask on Slack or ML about this problem (required/readonly).

Reply:
@britter 
@eolivelli 
hey guys, can you please check the late history how it was with this parameter? Maybe it was intended that `required=true` but `readonoly=false`. Perhaps the user could change the path to test `*.class`.